intermediate_source/fx_conv_bn_fuser.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+"""
+(beta) Building a Conv/BN fuser in FX
+*******************************************************
+**Author**: `Horace He <https://github.com/chillee>`_
+
+In this tutorial, we are going to use FX to do the following:
+
+1) Find patterns of conv/batch norm.
+2) Fold the batch norm statistics into the convolution weights.
+
+Despite this being a fairly trivial graph rewrite, this has been surprisingly
+difficult to do in PyTorch for quite some time.
+
+We will be building the fuser that exists here:
+https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/fuser.py
+
+"""
+
+
+######################################################################
+# First, let's get some imports out of the way (we will be using all
+# of these later in the code).
+
+from typing import Type, Dict, Any, Tuple, Iterable
+import copy
+import torch.fx as fx
+import torch
+import torch.nn as nn
+
+######################################################################
+# For this tutorial, we are going to create a model consisting of convolutions
+# and batch norms. Note that this model has some tricky components - some of
+# the modules are hidden within sequentials and one of the modules is wrapped
+# inside of another PyTorch module.
+
+class Wrapper(nn.Module):
```

Comment:
From linear scanning, it's not clear to me what this code example is. Is `M` the model we're testing on? What is `Wrapper` and why is it needed? The lead-in sentence isn't clear on whether we're matching on `Sequential` or otherwise

Reply:
It's not needed, it's just meant to making the matching a little bit harder. I'll rename `Wrapper` to `WrappedBatchnrom`